### PR TITLE
feat: Write asset overview report as csv instead of json (DEV-3300)

### PR DESCRIPTION
**Pull request by seakayone**
changes the report to csv:

```csv
shortcode,totalNrOfAssets,nrOfStillImageAssets,nrOfMovingImageAssets,nrOfOtherAssets,sizeOfStillImageOriginals,sizeOfStillImageDerivatives,sizeOfMovingImageOriginals,sizeOfMovingImageDerivatives,sizeOfMovingImageKeyframes,sizeOfOtherOriginals,sizeOfOtherDerivatives
0002,4,2,1,1,"1,11 MiB","1007,35 KiB","30,03 MiB","30,03 MiB","115,27 KiB","3,00 B","3,00 B"
0001,5,5,0,0,"54,48 KiB","14,91 KiB","0,00 B","0,00 B","0,00 B","0,00 B","0,00 B"
```

as a table

| shortcode | totalNrOfAssets | nrOfStillImageAssets | nrOfMovingImageAssets | nrOfOtherAssets | sizeOfStillImageOriginals | sizeOfStillImageDerivatives | sizeOfMovingImageOriginals | sizeOfMovingImageDerivatives | sizeOfMovingImageKeyframes | sizeOfOtherOriginals | sizeOfOtherDerivatives |
|-----------|-----------------|----------------------|-----------------------|-----------------|---------------------------|-----------------------------|-----------------------------|------------------------------|----------------------------|----------------------|------------------------|
| 0002      | 4               | 2                    | 1                     | 1               | 1,11 MiB                  | 1007,35 KiB                 | 30,03 MiB                   | 30,03 MiB                    | 115,27 KiB                 | 3,00 B               | 3,00 B                 |
| 0001      | 5               | 5                    | 0                     | 0               | 54,48 KiB                 | 14,91 KiB                   | 0,00 B                      | 0,00 B                       | 0,00 B                      | 0,00 B               | 0,00 B                 |

### CODE REVIEW

File: src/main/scala/swiss/dasch/domain/CsvService.scala
Comment (BalduinLandolt) [inline src/main/scala/swiss/dasch/domain/CsvService.scala:1]: **thought (non-blocking):** `CsvService`, `CsvRow` etc. are very generic names, on the other side, the actual implementation is specific to the asset overview workflow. Maybe the naming or the package structure could reflect that?

**Review by BalduinLandolt — APPROVED** [review]

File: src/main/scala/swiss/dasch/domain/CsvService.scala
Comment (seakayone) [reply src/main/scala/swiss/dasch/domain/CsvService.scala:1]: I have renamed `CsvRow` to `AssetOverviewReportCsvRow`.
WDYT?

Comment (BalduinLandolt) [reply src/main/scala/swiss/dasch/domain/CsvService.scala:1]: A bit lengthy, but at least clear what it does. I don't insist on this change, but I like it, thanks!

File: src/main/scala/swiss/dasch/domain/CsvService.scala [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    ZIO.acquireRelease(acquire)(release)
+  }
+
+  def writeReportToCsv(report: Seq[AssetOverviewReport], path: Path): Task[Path] =
+    ZIO.scoped {
+      for {
+        writer <- createWriter(path)
+        _      <- ZIO.succeed(writer.writeRow(AssetOverviewReportCsvRow.headerRow))
+        _ <- ZIO.foreachDiscard(report)(rep =>
+               ZIO.succeed(writer.writeRow(AssetOverviewReportCsvRow.fromReport(rep).toList))
```
Comment (mpro7) [inline src/main/scala/swiss/dasch/domain/CsvService.scala:111]: Weird formatting. Do we have `scalafmt` check here or should we have one in ingest too?

**Review by siers — APPROVED** [review]

File: src/main/scala/swiss/dasch/domain/CsvService.scala (hunk 1 above)
Comment (seakayone) [reply src/main/scala/swiss/dasch/domain/CsvService.scala:111]: https://github.com/dasch-swiss/dsp-ingest/blob/main/.github/workflows/ci.yml#L33

**Pull request merged**